src/js/letters/containers/AddressSection.jsx

```diff
@@ -43,45 +63,130 @@ export class AddressSection extends React.Component {
     }
   }
 
-  handleSave = () => {
-    this.setState({ isEditingAddress: false });
+  /**
+   * Runs all the validations against the address passed as a prop for a given field.
+   *
+   * @param {String} fieldName             The name of the address field to validate.
+   *                                         Maps to the fieldValidations key.
+   * @param {Object} fullAddress            Contains the full mailing address.
+   * @param {Boolean} ignoreShouldValidate  Because we'll need to update the error
+   *                                         messages on multiple fields sometimes,
+   *                                         we need to run validations on all fields.
+   *                                         This ensures that we only run validation
+   *                                         if the field has been modified.
+   *                                         saveAddress will need to validate all
+   *                                         fields regardless of whether they've been
+   *                                         modified.
+   * @return {String|undefined}             If there's a validation error, return the
+   *                                         error message. If not, return undefined.
+   */
+  validateField = (fieldName, fullAddress, ignoreShouldValidate = false) => {
+    // Only validate the field if it's been modified
+    if (!this.state.shouldValidate[fieldName] || ignoreShouldValidate) {
+      return undefined;
+    }
+
+    const validations = fieldValidations[fieldName];
+    // If there is no validations array for that field, assume it has no validations
+    if (!validations) {
+      return undefined;
+    }
+
+    let errorMessage = false;
+    for (let i = 0; i < validations.length; i++) {
+      // this.props.value = address
+      errorMessage = validations[i](fullAddress[fieldName], fullAddress);
+      if (typeof errorMessage === 'string') {
+        return errorMessage;
+      }
+    }
+
+    // All validations passed; there are no error messages to report
+    return undefined;
+  }
+
+  /**
+   * Runs validation for all fields, returning a complete errorMessages object.
+   *
+   * @return {Object}  Holds all the error messages for all the fields that have them.
+   */
+  validateAll = (address = this.state.editableAddress) => {
+    const errorMessages = {};
+    Object.keys(fieldValidations).forEach((fieldName) => {
+      errorMessages[fieldName] = this.validateField(fieldName, address);
+    });
+
+    return errorMessages;
+  }
 
+  // TODO: Make sure this doesn't allow us to save the address if there are validation errors
```

Comment:
Could just disable the button if there are error messages instead of touching the `handleSave` logic, right?

Reply:
Yeah, that's an option. We're not doing that anywhere else as far as I know, so I'd have to rope design in on that decision. I'll write another ticket for it.